### PR TITLE
Add codecov coverage tests

### PR DESCRIPTION
**Pull request by ohm314**
**Description**

Use gcov/lcov to generate coverage reports on running `make test` and report coverage in codecov.io

**Use certain branches for the SimulationStack CI**

CI_BRANCHES:NEURON_BRANCH=master,

### CODE REVIEW

File: .github/workflows/coreneuron-ci.yml [hunk 1]
```diff
@@ -161,3 +161,30 @@ jobs:
 #           BRANCH: gh-pages # The branch the action should deploy to.
 #           FOLDER: ${{runner.workspace}}/nrn/build/docs # The folder the action should deploy.
 #           CLEAN: false # Automatically remove deleted files from the deploy branch
+  coverage:
+    runs-on: ubuntu-16.04
+    name: "Coverage Test"
+    env:
+      INSTALL_DIR: install
```
Comment (alexsavulescu) [inline .github/workflows/coreneuron-ci.yml:168]: not needed

**Review by alexsavulescu — COMMENTED** [review]
Overall LGTM, happy with having a separate job for now.

**Comment by pramodk** [comment]
Apart from Alex comments, this is good! this is going to be very useful!

>  btw, we could add a badge 💪

and yes, badge please! :)

**Comment by pramodk** [comment]
I am looking at the build : https://github.com/BlueBrain/CoreNeuron/pull/478/checks?check_run_id=1751948118
where we have:

 -> View reports at https://codecov.io/github/BlueBrain/CoreNeuron/commit/e99feb12aa4b35823c403c2e64e5a1efc5c3c0dc

Clicking there and looking at specific file gives:

image

Seems like others have reported similar issue: https://community.codecov.io/t/code-file-coverage-report-github-api-forbidden-is-the-path-a-relative-path-in-your-repository/615/4

**Comment by ohm314** [comment]
Now it seems to work
but what is strange is that codecov still thinks the CI is ongoing, which is why it did not update the "checks" list in the PR here with direct links to the report

**Review by pramodk — APPROVED** [review]
LGTM! 
lets see if reports are updated "sooner" after merge.

**Pull request merged**